src/Symfony/Component/Console/DependencyInjection/AddConsoleCommandPass.php

```diff
@@ -55,7 +55,7 @@ public function process(ContainerBuilder $container)
 
             $commandId = 'console.command.'.strtolower(str_replace('\\', '_', $class));
 
-            if (!isset($tags[0]['command'])) {
+            if (!isset($tags[0]['command']) && !$commandName = $class::getDefaultName()) {
```

Comment:
an empty string would give the expected unexpected result :) `"0"` could be a valid name. Anyway... this is nitpicking; fine asis for me :+1: